### ISSUE TITLE
Restore never starts on version 4.50.2 on Ubuntu 22.04

### ISSUE THREAD

**Issue by intrepidsilence** [homebridge/homebridge-config-ui-x#1457]
### Describe The Bug

When trying to Restore from Backup, the restore never starts and we get a spinning wheel forever. Only seems to affect 4.50.2 as rolling back to 4.50.1 or earlier allows the restore to function as expected.  Issue was reported on the Homebridge git: [https://github.com/homebridge/homebridge/issues/3305](https://github.com/homebridge/homebridge/issues/3305)

### Logs

```text
No logs are written during the occurrence of this issue.
```

### Homebridge UI Version

4.50.2

### Homebridge Version

1.6.0

### Node.js Version

18.14.0

### Operating System

Ubuntu / Debian (or a variant)

### Environment Info

- [ ] Using Docker?
- [ ] Using Hyper-V?
- [X] Using hb-service?

### Raspberry Pi Model

None

**Comment by shanemcw** [comment]
I also see this behavior on
4.50.2

Ubuntu Jammy Jellyfish (22.04.2 LTS)
1.6.0 homebridge
v18.13.0 nodejs

/var/lib/homebridge
/var/lib/homebridge/config.json
/var/lib/homebridge/node_modules

**Comment by shanemcw** [comment]
Bump. Not stale.

**Comment by Biggsylad** [comment]
+1 Same issue

**Comment by NorthernMan54** [comment]
Spent some time this am on this issue, and was able to recreate the issue.  I created a virtual box image of Ubuntu Jammy Jellyfish (22.04.2 LTS) using the server os, and installed Homebridge per the wiki.

I then connected to the homebridge ui, and selected the restore from backup option, selected a backup image from another machine, and it got stuck.

I then went and downgraded config-ui-x with `npm install -g homebridge-config-ui-x@4.50.1` from the hb-shell prompt in the  /home/homebridge directory.

Restarted homebridge, and the restore was successful.

**Comment by NorthernMan54** [comment]
Spent more time this morning on this issue, and was able to recreate on a RPI 2 running bullseye

FYI - to reset the environment between tests

`sudo apt-get remove -y homebridge; sudo apt-get purge -y homebridge ; sudo apt-get install -y homebridge ; sudo hb-service logs`

**Comment by NorthernMan54** [comment]
While we work thru this issue, these are the command line ( shell ) commands to downgrade the bundled version on config-ui-x so you can perform the restore

From a command line on your environment

```
sudo hb-shell
/opt/homebridge/lib/node_modules/homebridge-config-ui-x/upgrade-install.sh 4.50.1 /opt/homebridge
exit
sudo hb-service restart
sudo hb-service logs
```

This will install the older version so you can perform the restore

**Comment by NorthernMan54** [comment]
Spent more time investigating this issue, and am getting baffled by the root cause.

If I setup the environment from scratch ie

```
sudo apt-get remove -y homebridge; sudo apt-get purge -y homebridge ; sudo apt-get install -y homebridge ; sudo hb-service logs
```

The issue occurs

Then if I go into the environment and re-install the latest homebridge-config-ui-x aka 4.50.2

ie

```
npm install -g homebridge-config-ui-x@latest
```

The issue does not occur, which means it is not an issue with the code within this package but something else.  In this setup, Homebridge-config-ui-x is installed by the homebridge-apt-pkg.  I looked at the homebridge-apt-pkg and it is just installing the latest

https://github.com/homebridge/homebridge-apt-pkg/blob/9324c19145ab176deff7637f4817e9624908d6d6/build.sh#L48

**Comment by NorthernMan54** [comment]
I looked into the `homebridge-apt-pkg` deb package deeper and see that it contains a pre-built directory structure containing homebridge, node, npm and homebridge-config-ui-x including all the dependencies, so my previous comment is not correct.  The file I referenced previously is used as part of the process to create the package, and not during the installation process.

With that said, I grabbed a copy of the current homebridge apt pkg, homebridge_1.0.33_amd64.deb, exploded it, ran npm update against the homebridge-config-ui-x within the package, then created an updated package.

I then installed the updated package and could not recreate the issue.  So this confirms my earlier finding that their is an issue within the dependencies of homebridge-config-ui-x.

So to resolve the issue we need to just re-release homebridge-apt-pkg, and this will update the dependencies and resolve the issue for new installations going forward.

For existing installations, re-installing homebridge-config-ui-x will resolve the issue as this will update the dependencies.

ie

```
sudo hb-shell
/opt/homebridge/lib/node_modules/homebridge-config-ui-x/upgrade-install.sh 4.50.1 /opt/homebridge
exit
sudo hb-service restart
sudo hb-service logs
```
